synapseclient/models/wiki.py

```diff
@@ -1 +1,804 @@
-# TODO
+"""Script to work with Synapse wiki pages."""
+
+import gzip
+import os
+import shutil
+import uuid
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Union
+
+from synapseclient import Synapse
+from synapseclient.api.wiki_service import (
+    delete_wiki_page,
+    get_attachment_handles,
+    get_attachment_preview_url,
+    get_attachment_url,
+    get_markdown_url,
+    get_wiki_header_tree,
+    get_wiki_history,
+    get_wiki_order_hint,
+    get_wiki_page,
+    post_wiki,
+    put_wiki_order_hint,
+    put_wiki_page,
+    put_wiki_version,
+)
+from synapseclient.core.async_utils import async_to_sync, otel_trace_method
+from synapseclient.core.upload.upload_functions_async import upload_file_handle
+from synapseclient.core.utils import delete_none_keys
+
+
+@dataclass
+@async_to_sync
+class WikiOrderHint:
+    """
+    A WikiOrderHint contains the order hint for the root wiki that corresponds to the given owner ID and type.
+
+    Attributes:
+        owner_id: The ID of the owner object (e.g., entity, evaluation, etc.).
+        owner_object_type: The type of the owner object.
+        id_list: The list of sub wiki ids that in the order that they should be placed relative to their siblings.
+        etag: The etag of this object.
+    """
+
+    owner_id: Optional[str] = None
+    """The ID of the owner object (e.g., entity, evaluation, etc.)."""
+
+    owner_object_type: Optional[str] = None
+    """The type of the owner object."""
+
+    id_list: List[str] = field(default_factory=list)
+    """The list of sub wiki ids that in the order that they should be placed relative to their siblings."""
+
+    etag: Optional[str] = field(default=None, compare=False)
+    """The etag of this object."""
+
+    def fill_from_dict(
+        self,
+        wiki_order_hint: Dict[str, Union[str, List[str]]],
+    ) -> "WikiOrderHint":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            wiki_order_hint: The response from the REST API.
+
+        Returns:
+            The WikiOrderHint object.
+        """
+        self.owner_id = wiki_order_hint.get("ownerId", None)
+        self.owner_object_type = wiki_order_hint.get("ownerObjectType", None)
+        self.id_list = wiki_order_hint.get("idList", [])
+        self.etag = wiki_order_hint.get("etag", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Wiki_Order_Hint: {self.owner_id}"
+    )
+    async def get_async(
+        self,
+        *,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> "WikiOrderHint":
+        """
+        Get the order hint of a wiki page tree asynchronously.
```

Comment:
```suggestion
        Get the order hint of a wiki page tree.
```

We don't need to mention `asynchronously` in the docstring